e2e_tests/test_visual_regression.py

```diff
@@ -3,47 +3,145 @@
 
 
 @pytest.mark.parametrize(
-    "url,screenshot_file",
+    "url,screenshot_file,viewport",
     [
-        ("/how-to-use-this-service", "how_to_use_this_service.png"),
-        ("/terms-of-use", "terms_of_use.png"),
-        ("/privacy", "privacy.png"),
-        ("/cookies", "cookies.png"),
-        ("/accessibility", "accessibility.png"),
-        ("/signed-out", "signed_out.png"),
-        ("/", "start.png"),
-        ("/browse", "browse.png"),
+        (
+            "/how-to-use-this-service",
+            "how_to_use_this_service.png",
+            {"width": 1280, "height": 4000},
+        ),
+        ("/terms-of-use", "terms_of_use.png", {"width": 1280, "height": 4000}),
+        ("/privacy", "privacy.png", {"width": 1280, "height": 4000}),
+        ("/cookies", "cookies.png", {"width": 1280, "height": 4000}),
+        (
+            "/accessibility",
+            "accessibility.png",
+            {"width": 1280, "height": 4000},
+        ),
+        ("/signed-out", "signed_out.png", {"width": 1280, "height": 4000}),
+        ("/", "start.png", {"width": 1280, "height": 4000}),
+        ("/browse", "browse.png", {"width": 1280, "height": 4000}),
         (
             "/browse/transferring_body/8ccc8cd1-c0ee-431d-afad-70cf404ba337",
             "browse_transferring_body.png",
+            {"width": 1280, "height": 4000},
         ),
         (
             "/browse/series/8bd7ad22-90d1-4c7f-ae00-645dfd1987cc",
             "browse_series.png",
+            {"width": 1280, "height": 4000},
         ),
         (
             "/browse/consignment/a03363ac-7e7b-4b92-817e-72ba6423edd5",
             "browse_consignment.png",
+            {"width": 1280, "height": 4000},
+        ),
+        (
+            "/search_results_summary?query=a",
+            "search_results_summary.png",
+            {"width": 1280, "height": 4000},
         ),
-        ("/search_results_summary?query=a", "search_results_summary.png"),
         (
             "/search/transferring_body/8ccc8cd1-c0ee-431d-afad-70cf404ba337?query=a&sort=series-asc&search_filter=test",
             "search_transferring_body.png",
+            {"width": 1280, "height": 4000},
+        ),
+        (
+            "/record/41f94132-dbdf-43e4-a327-cc5bae432f98",
+            "record.png",
+            {"width": 1280, "height": 4000},
         ),
-        ("/record/41f94132-dbdf-43e4-a327-cc5bae432f98", "record.png"),
     ],
 )
 def test_css_no_visual_regression(
-    url, screenshot_file, aau_user_page: Page, assert_snapshot
+    url, screenshot_file, viewport, aau_user_page: Page, assert_snapshot
+):
+    """
+    Given a page in the AYR webapp
+    When an authenticated user navigates to it
+    Then it should look as expected on a desktop viewport
+
+    If any of these tests break due to intended changes to the design of the page,
+    run pytest with `--update-snapshots --headed` flags to update the stored screenshot
+    """
+    aau_user_page.set_viewport_size(viewport)
```

Comment:
so are we certain that setting the viewport size will fix the flakiness of these tests? I guess we'll see as we run them in the coming week on different machines etc

Reply:
I think they should be better, it should at minimum remove any failures due to different dimensions. The maximum dimensions I could see was around 3500 height for static content pages so 4000 (the current desktop config) should never be reached.

Reply:
okay, did you figure out why they were being rendered with different dimensions if we werent specifying anything differently? Is that just some sort of inherent randomness of browsers?

And ah yeah I didnt even consider that, but if we're sure we shouldn't exceed that then seems fine

Reply:
The default viewport for playwright is 1280x720 which it should have been using, whether it was due to the viewport or another issue I'm not 100% sure yet. If we still have issues we may need to configure the threshold for allowed diffs within screenshots.

This playwright discussion may also be useful:
https://github.com/microsoft/playwright/issues/8161

Threshold setting:
https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-screenshot-2